### PR TITLE
Fix behavior of entry_float_type flag in import_vcf

### PR DESCRIPTION
**Pull request by tmwong2003**
This PR fixes an issue in which the `entry_float_type` flag in `import_vcf` affects all floating-point fields in a VCF, including fields defined in `INFO` headers. We should have restricted it to affect only fields defined in `FORMAT` headers.

### CODE REVIEW

File: hail/src/main/scala/is/hail/io/vcf/LoadVCF.scala [hunk 1]
```diff
@@ -1089,15 +1133,17 @@ object LoadVCF {
...
-
-    val baseType = (line.getType, isCall) match {
-      case (VCFHeaderLineType.Integer, false) => TInt32()
-      case (VCFHeaderLineType.Float, false) => entryFloatType
-      case (VCFHeaderLineType.String, true) => TCall()
-      case (VCFHeaderLineType.String, false) => TString()
-      case (VCFHeaderLineType.Character, false) => TString()
-      case (VCFHeaderLineType.Flag, false) => TBoolean()
-      case (_, true) => fatal(s"Can only convert a header line with type 'String' to a call type. Found '${ line.getType }'.")
+    val isEntry = line.getKey == "FORMAT"
```
Comment (tpoterba) [inline hail/src/main/scala/is/hail/io/vcf/LoadVCF.scala]: might be a bit cleaner to change `entryFloatType` to just `floatType` and change the following line in `parseHeader`:

    val (infoSignature, infoAttrs, infoFlagFields) = headerSignature(infoHeader, callFields, entryFloatType)

to

    val (infoSignature, infoAttrs, infoFlagFields) = headerSignature(infoHeader, callFields, TFloat64())

**Comment by tpoterba** [comment]
we recently locked down our CI to run PRs only from the development team -- we intend to add a feature to our CI to let us run the test suite on a specific PR and SHA, but don't have that feature yet... 😦 

@cseed 	@danking this isn't hard, right?

**Comment by danking** [comment]
I should be able to do it today if I get my laptop fixed

File: hail/src/main/scala/is/hail/io/vcf/LoadVCF.scala (hunk 1 above)
Comment (tmwong2003) [reply hail/src/main/scala/is/hail/io/vcf/LoadVCF.scala]: Fixed - see commit 26fc465.

**Comment by tmwong2003** [comment]
> I should be able to do it today if I get my laptop fixed

Is there a way to bump a CI run on this PR? Thank you.

**Comment by cseed** [comment]
Hey @tmwong2003, we recently changed our CI setup and there's a little more work to do to test external PRs.  I'll try to have someone finish this in the next couple of days.  Thanks for your patience.

**Comment by tmwong2003** [comment]
Thanks Cotton. CI management is never easy at the best of times so no worries.

- T

-- 
Theodore Wong <tmw@tmwong.org>
http://www.tmwong.org

**Comment by cseed** [comment]
@tmwong2003 OK, some progress on the CI front.  Thanks for your patience.

A team member needs to kick off the CI job.  The assigned reviewer will be responsible for that.  Right now, the tests involve some sensitive tokens which need more work to be protected, so the CI logs aren't public yet.  Again, the assigned reviewer should be able to share the relevant part of the logs for failures, etc.

@tpoterba I kicked off the build.  Can you take another look, it looks like the comments were addressed.

Finally, it looks like the PR history is a bit tangled.  Is it possible to clean it up so we can a reasonable commit message?  Thanks!

File: hail/src/main/scala/is/hail/io/vcf/LoadVCF.scala [hunk 2]
```diff
@@ -1156,10 +1202,10 @@ object LoadVCF {
       .toMap
 
     val infoHeader = header.getInfoHeaderLines
-    val (infoSignature, infoAttrs, infoFlagFields) = headerSignature(infoHeader, callFields, entryFloatType)
+    val (infoSignature, infoAttrs, infoFlagFields) = headerSignature(infoHeader, callFields, floatType)
```
Comment (tpoterba) [inline hail/src/main/scala/is/hail/io/vcf/LoadVCF.scala]: this should be TFloat64(), I think

File: hail/src/main/scala/is/hail/io/vcf/LoadVCF.scala (hunk 1 above)
Comment (tpoterba) [reply hail/src/main/scala/is/hail/io/vcf/LoadVCF.scala]: I don't think you need this logic anymore -- if you change the info field headerSignature call to use TFloat64, then the float type passed in here is the one you should use.

**Comment by danking** [comment]
@tmwong2003 it looks like you deleted the branch connected to this PR https://github.com/tmwong2003/hail/branches was that intentional? Our CI is a bit confused now.

**Comment by tmwong2003** [comment]
Rats. I re-forked from the master hail repo because I’d managed to scramble our fork of it. Nuts. Please close this PR and I’ll resubmit.

Sorry…

-- 
Theodore Wong <tmw@tmwong.org>
http://www.tmwong.org

File: hail/src/main/scala/is/hail/io/vcf/LoadVCF.scala (hunk 2 above)
Comment (tmwong2003) [reply hail/src/main/scala/is/hail/io/vcf/LoadVCF.scala]: OK, I fixed that and simplified the other change in `headerField`.

**Comment by tmwong2003** [comment]
Closed this PR because of forked repo damage at our end - resubmitted as #6378.

**Pull request closed without merging**